File: app/src/main/java/dev/ragnarok/fenrir/model/Owner.java
package dev.ragnarok.fenrir.model;

import android.os.Parcel;

import dev.ragnarok.fenrir.module.parcel.ParcelNative;

public abstract class Owner extends AbsModel implements ParcelNative.ParcelableNative {

    @OwnerType
    private final int ownerType;

    protected Owner(int ownerType) {
        this.ownerType = ownerType;
    }

    public Owner(Parcel in) {
        super(in);
        //noinspection ResourceType
        ownerType = in.readInt();
    }

    public Owner(ParcelNative in) {
        //noinspection ResourceType
        ownerType = in.readInt();
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        super.writeToParcel(parcel, i);
        parcel.writeInt(ownerType);
    }

    @Override
    public void writeToParcelNative(ParcelNative parcel) {
        parcel.writeInt(ownerType);
    }

    @OwnerType
    public int getOwnerType() {
        return ownerType;
    }

    public int getOwnerId() {
        throw new UnsupportedOperationException();
    }

    public String getDomain() {
        throw new UnsupportedOperationException();
    }

    public String getMaxSquareAvatar() {
        throw new UnsupportedOperationException();
    }

    public String getOriginalAvatar() {
        throw new UnsupportedOperationException();
    }

    public String get100photoOrSmaller() {
        throw new UnsupportedOperationException();
    }

    public String getFullName() {
        throw new UnsupportedOperationException();
    }

    public boolean isVerified() {
        throw new UnsupportedOperationException();
    }

    public boolean isDonated() {
        throw new UnsupportedOperationException();
    }
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/model/Owner.kt
package dev.ragnarok.fenrir.model

import android.os.Parcel
import dev.ragnarok.fenrir.module.parcel.ParcelNative

abstract class Owner : AbsModel, ParcelNative.ParcelableNative {
    @OwnerType
    val ownerType: Int

    protected constructor(ownerType: Int) {
        this.ownerType = ownerType
    }

    protected constructor(`in`: Parcel) : super(`in`) {
        ownerType = `in`.readInt()
    }

    protected constructor(`in`: ParcelNative) {
        ownerType = `in`.readInt()
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        super.writeToParcel(parcel, i)
        parcel.writeInt(ownerType)
    }

    override fun writeToParcelNative(dest: ParcelNative) {
        dest.writeInt(ownerType)
    }

    open val ownerId: Int
        get() {
            throw UnsupportedOperationException()
        }
    open val domain: String?
        get() {
            throw UnsupportedOperationException()
        }
    open val maxSquareAvatar: String?
        get() {
            throw UnsupportedOperationException()
        }
    open val originalAvatar: String?
        get() {
            throw UnsupportedOperationException()
        }

    open fun get100photoOrSmaller(): String? {
        throw UnsupportedOperationException()
    }

    open val fullName: String?
        get() {
            throw UnsupportedOperationException()
        }
    open val isVerified: Boolean
        get() {
            throw UnsupportedOperationException()
        }
    open val isDonated: Boolean
        get() {
            throw UnsupportedOperationException()
        }
}